pkg/deploy/generator/scripts/gatewayVMSS.sh

```diff
@@ -316,7 +318,20 @@ echo "Download \$COMPONENT credentials"
 
 TEMP_DIR=\$(mktemp -d)
 export AZURE_CONFIG_DIR=\$(mktemp -d)
-az login -i --allow-no-subscriptions
+
+echo "Logging into Azure..."
+RETRIES=5
+while [ "\$RETRIES" -gt 0 ]; do
+    if az login -i --allow-no-subscriptions
+    then
+        echo "az login successful"
+        break
+    else
+        echo "az login failed. Retrying..."
+        let RETRIES-=1
+        sleep 5
+    fi
+done
```

Comment:
I've actually improved the success rate of this command by including `After` and `Wants` for `network-online.target` in the systemd timers in my latest iteration, but since they only fire once an hour, having retries is still a good idea due to the sporadic errors, like "Unable to load MSI", we do encounter.